docker-app/qfieldcloud/core/serializers.py

```diff
@@ -535,3 +539,44 @@ class LatestPackageSerializer(serializers.Serializer):
     package_id = serializers.UUIDField()
     packaged_at = serializers.DateTimeField()
     data_last_updated_at = serializers.DateTimeField()
+
+
+class TeamSerializer(serializers.ModelSerializer):
+    organization = serializers.StringRelatedField(source="team_organization")
+
+    class Meta:
+        model = Team
+        fields = ("username", "organization")
+
+
+class TeamMemberSerializer(serializers.ModelSerializer):
+    member = serializers.CharField()
+
+    def to_internal_value(self, data):
+        validated_data = super().to_internal_value(data)
+        email_or_username = data.get("member")
+        existing_username = self.context["view"].kwargs.get("member_username")
+
+        if existing_username is not None:
+            if not email_or_username:
+                raise serializers.ValidationError(
+                    {"member": "Username must be provided"}
+                )
+
+            try:
+                if User.objects.fast_search(email_or_username) > 0:
+                    raise serializers.ValidationError(
+                        {
+                            "member": f"A user with username '{email_or_username}' already exists."
+                        }
+                    )
```

Comment:
You are not adding a user, I guess the message should be different.

```suggestion
                    raise serializers.ValidationError(
                        {
                            "member": f"A member with username '{email_or_username}' already exists."
                        }
                    )
```

Reply:
This comments is answering all the comments above related to function internal_value:

The idea is the following:
1. check if the username of the member that is going to be added already exists in another member
2. check if member is indeed a user (this might be overkill due to an existing check via permissions in the view(???)
3. otherwise use the fast_search to obtain the user

```
  try:
            existing_user = User.objects.fast_search(email_or_username)
            
            if TeamMember.objects.filter(member=existing_user):
                raise serializers.ValidationError(
                        {
                            "member": f"A member with username '{email_or_username}' already exists."
                        }
                )
            
            validated_data["member"] = existing_user
        
        except User.DoesNotExist:  
            
            raise serializers.ValidationError(
                        {
                            "member": f"A user with username '{email_or_username}' does not exists."
                        }
            )
        
        return validated_data
```

Reply:
What happens if we don't explicitly handle the `DoesNotExist` error, or we attempt to break the uniqueness of the members in the organization? Do we get a proper error or just error 500? What is the HTTP response body?

If we get 500, then we need to handle it. Your proposal is good.
 
Note that the messages are weird, you can get a validation message like: "A member with username user@email.com already exists". Just say 'The team member "user@email.com" already exists."  We always put placeholders in double quotes in user strings.